Utils/azuretools-core/src/com/microsoft/azure/toolkit/lib/common/operation/AzureOperationRef.java

```diff
@@ -24,25 +24,30 @@
 
 import lombok.Builder;
 import lombok.Getter;
-import lombok.Setter;
 
 import java.lang.reflect.Method;
+import java.util.Objects;
 
 @Getter
-@Setter
 @Builder
 public class AzureOperationRef {
-    private Method method;
-    private String[] paramNames;
-    private Object[] paramValues;
-    private Object instance;
+    private final Method method;
+    private final String[] paramNames;
+    private final Object[] paramValues;
+    private final Object instance;
 
     @Override
     public boolean equals(final Object obj) {
         if (!(obj instanceof AzureOperationRef)) {
             return false;
         }
         final AzureOperationRef operation = (AzureOperationRef) obj;
-        return operation.getMethod() == this.method || operation.getMethod().equals(this.method);
+        return operation.getMethod() == this.method || Objects.equals(operation.getMethod(), this.getMethod());
```

Comment:
Objects.equals(operation.getMethod(), this.getMethod()) is enough

Reply:
I checked `operation.getMethod() == this.method` first since it's faster and the implementation of `equals` in `Method` is some complex. 
In our cases, the checked obj are mostly the same one. so use `==` can be faster.
![image](https://user-images.githubusercontent.com/69189193/103391553-0732b880-4b55-11eb-94d4-588890d75cbf.png)

Reply:
you can check the code of Objects.equals